src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsResponseDetails.java

```diff
@@ -108,6 +77,67 @@ public static List<FeedbackResponseAttributes.UpdateOptions> getUpdateOptionsFor
         return updateOptions;
     }
 
+    /**
+     * Checks whether update is needed.
+     * @param responses responses to one feedback question, from one giver
+     * @param maxRank the maximum rank in each response
+     * @return boolean deciding if update is needed
+     */
+    private static boolean needsUpdate(List<FeedbackResponseAttributes> responses, int maxRank) {
+        FeedbackResponseDetails details;
+        FeedbackRankRecipientsResponseDetails responseDetails;
+        int answer;
+        boolean isUpdateNeeded = false;
+
+        for (FeedbackResponseAttributes response : responses) {
+            details = response.getResponseDetails();
+            if (!(details instanceof FeedbackRankRecipientsResponseDetails)) {
+                continue;
+            }
+            responseDetails = (FeedbackRankRecipientsResponseDetails) details;
+            answer = responseDetails.getAnswer();
+            if (answer > maxRank) {
```

Comment:
To reduce unnecessary local variables, I believe these 2 lines can be simplified to:
```
if (responseDetails.getAnswer() > maxRank) {
```